bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/internal/AudioServlet.java

```diff
@@ -159,58 +163,171 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
-    public String serve(FixedLengthAudioStream stream, int seconds) {
+    public String serve(AudioStream stream, int seconds) {
+        try {
+            return serve(stream, seconds, null);
+        } catch (IOException e) {
+            logger.warn("Cannot precache the audio stream to serve it", e);
+            return getRelativeURL("error");
+        }
+    }
+
+    @Override
+    public String serve(AudioStream stream, int seconds, @Nullable Runnable callBack) throws IOException {
+
         String streamId = UUID.randomUUID().toString();
-        multiTimeStreams.put(streamId, stream);
-        streamTimeouts.put(streamId, System.nanoTime() + TimeUnit.SECONDS.toNanos(seconds));
+        ClonableAudioStream clonableAudioStream = null;
+        if (stream instanceof ClonableAudioStream clonableAudioStreamDetected) {
+            clonableAudioStream = clonableAudioStreamDetected;
+        } else { // we prefer a ClonableAudioStream, but we can try to make one
+            clonableAudioStream = createClonableInputStream(stream, streamId);
+        }
+        long timeOut = System.nanoTime() + TimeUnit.SECONDS.toNanos(seconds);
+        StreamServed streamToServe = new StreamServed(streamId, clonableAudioStream, new AtomicInteger(),
+                new AtomicLong(timeOut), true, callBack);
+        servedStreams.put(streamId, streamToServe);
+
+        // try to clean, or a least launch the periodic cleanse:
+        removeTimedOutStreams();
+
         return getRelativeURL(streamId);
     }
 
-    Map<String, FixedLengthAudioStream> getMultiTimeStreams() {
-        return Collections.unmodifiableMap(multiTimeStreams);
+    private ClonableAudioStream createClonableInputStream(AudioStream stream, String streamId) throws IOException {
+        byte[] dataBytes = stream.readNBytes(ONETIME_STREAM_BUFFER_MAX_SIZE + 1);
+        ClonableAudioStream clonableAudioStreamResult;
+        if (dataBytes.length <= ONETIME_STREAM_BUFFER_MAX_SIZE) {
+            // we will use an in memory buffer to avoid disk operation
+            clonableAudioStreamResult = new ByteArrayAudioStream(dataBytes, stream.getFormat());
+        } else {
+            // in memory max size exceeded, sound is too long, we will use a file
+            File tempFile = File.createTempFile(streamId, ".snd");
+            tempFile.deleteOnExit();
+            try (OutputStream outputStream = new FileOutputStream(tempFile)) {
+                // copy already read data to file :
+                outputStream.write(dataBytes);
+                // copy the remaining stream data to a file.
+                byte[] buf = new byte[8192];
+                int length;
+                // but with a limit
+                int fileSize = ONETIME_STREAM_BUFFER_MAX_SIZE + 1;
+                while ((length = stream.read(buf)) != -1 && fileSize < ONETIME_STREAM_FILE_MAX_SIZE) {
+                    outputStream.write(buf, 0, length);
+                    fileSize += length;
+                }
```

Comment:
With this code, the file can finally have a size bigger than ONETIME_STREAM_FILE_MAX_SIZE

Reply:
Fixed